include/libopencm3/stm32/fdcan.h

```diff
@@ -55,6 +40,7 @@ LGPL License Terms @ref lgpl_license
 #define FDCAN_FIFO1						1
 /**@}*/
 
+#define FDCAN_BLOCK_ID(can_base)		(((can_base) - CAN1)/(CAN2 - CAN1))
```

Comment:
are you sure this is always a simple op? and not pulling in divide? (not as big a deal on "big" cpus as on small, but still, you don't want to be dividing a lot here?

Reply:
Checked with godbolt, ARM32 GCC 10.2.1 none with -mcpu=cortex-m0 -mthumb -msoft-float -O0 yields only loads, stores and shifts/rotates. With -O2, it boils down to load, move, add and shift.

Reply:
yay! I don't always have time to godbolt everything myself :) thanks for doing that.